File: py/src/formulas/algebra.py
def solve_linear(a, b):
    if a == 0:
        raise ValueError("Coefficient 'a' cannot be zero.")
    return -b / a

def quadratic_roots(a, b, c):
    import math
    discriminant = b ** 2 - 4 * a * c
    if discriminant < 0:
        raise ValueError("No real roots.")
    root1 = (-b + math.sqrt(discriminant)) / (2 * a)
    root2 = (-b - math.sqrt(discriminant)) / (2 * a)
    return root1, root2

File: cs/src/formulas/algebra.cs
using System;

public static class Algebra
{
    public static double SolveLinear(double a, double b)
    {
        if (a == 0)
            throw new ArgumentException("Coefficient 'a' cannot be zero.");
        return -b / a;
    }

    public static (double, double) QuadraticRoots(double a, double b, double c)
    {
        double discriminant = b * b - 4 * a * c;
        if (discriminant < 0)
            throw new ArgumentException("No real roots.");
        double root1 = (-b + Math.Sqrt(discriminant)) / (2 * a);
        double root2 = (-b - Math.Sqrt(discriminant)) / (2 * a);
        return (root1, root2);
    }
}
